### ISSUE TITLE
CentOS 7 ppc64le on Power9 - Unsupported distribution?

### ISSUE THREAD

**Issue by PeS-hkfree** [NVIDIA/libnvidia-container#69]
I'm trying to install nvidia-docker2 on CentOS 7.7 on power 9 - ppc64le.  But when I follow the guide, yum is unable to install packeges - HTTPS 404 error, whe I directly go to the address I get error # Unsupported distribution! # Check https://nvidia.github.io/libnvidia-container . I tried to disable all gpg etc checks, change it to rhel7, still same error.
What I'm doing wrong, is nvidia-docker2 really supported on CentOS7 on ppc64le?

**Comment by dereklstinson** [comment]
Same thing is happening on the link https://nvidia.github.io/libnvidia-container/ubuntu18.04/amd64.

**Comment by retrack** [comment]
> Same thing is happening on the link https://nvidia.github.io/libnvidia-container/ubuntu18.04/amd64.

yes, please adjust;

**Comment by paulhutchings** [comment]
Any updates on this? [https://nvidia.github.io/libnvidia-container/ubuntu18.04/](https://nvidia.github.io/libnvidia-container/ubuntu18.04/) is still giving an error.

**Comment by nvjmayo** [comment]
Directory listing isn't enabled and will report the error message you see: 
`# Unsupported distribution! # Check https://nvidia.github.io/libnvidia-container
`

Here's an example of a URL that works, as described in the [Repository Configuration](https://nvidia.github.io/libnvidia-container) guide: https://nvidia.github.io/libnvidia-container/ubuntu18.04/libnvidia-container.list 

When you use `apt`, it is going to look for something like https://nvidia.github.io/libnvidia-container/ubuntu18.04/amd64/Packages.xz based on what you added to `/etc/apt/sources.list.d/libnvidia-container.list`. 

For the ppc64el I confirmed they exist with this script: 
`curl --head https://nvidia.github.io/libnvidia-container/{ubuntu16.04,ubuntu18.04,debian9}/ppc64el/Packages.xz`

You should see 200 OK results for all 3 distros, not 404 results. Please note that it's **ppc64el** not _ppc64le_. Update your sources.list.d if it's pointing to the wrong $ARCH

**Comment by mankeyboy** [comment]
@nvjmayo Can you give us a modified list of steps here for what exactly to do?  
I'm trying to get this running on RHEL7.6 with ppc64 and your comment above makes it seem like only Ubuntu and debian flavours are supported. Can you explain how to get this working for RHEL type distributions, especially with ppc64.
I tried `rhel7.6` and it gives me a 404 while the debian ones give 200 as you mentioned.

**Comment by PeS-hkfree** [comment]
As a unsecure fix I disabled gpg checks. My config for CentOS 7 is now this:
`[libnvidia-container]
name=libnvidia-container
baseurl=https://nvidia.github.io/libnvidia-container/centos7/$basearch
repo_gpgcheck=0
gpgcheck=0
enabled=1
gpgkey=https://nvidia.github.io/libnvidia-container/gpgkey
sslverify=0
sslcacert=/etc/pki/tls/certs/ca-bundle.crt

[nvidia-container-runtime]
name=nvidia-container-runtime
baseurl=https://nvidia.github.io/nvidia-container-runtime/centos7/$basearch
repo_gpgcheck=0
gpgcheck=0
enabled=1
gpgkey=https://nvidia.github.io/nvidia-container-runtime/gpgkey
sslverify=0
sslcacert=/etc/pki/tls/certs/ca-bundle.crt

[nvidia-docker]
name=nvidia-docker
baseurl=https://nvidia.github.io/nvidia-docker/centos7/$basearch
repo_gpgcheck=0
gpgcheck=0
enabled=1
gpgkey=https://nvidia.github.io/nvidia-docker/gpgkey
sslverify=0
sslcacert=/etc/pki/tls/certs/ca-bundle.crt
'

**Comment by qhaas** [comment]
I'm starting to wonder if support for EL7 (RHEL7, CentOS7, etc.) on ppc64le has been dropped, discovered while troubleshooting an issue with Vulkan over in Issue #55 that the EL7 ppc64le version is from 9/18, whereas the x86-64 version is from 1/20.  At least based on what `nvidia-container-cli --version` is returning.

**Comment by RenaudWasTaken** [comment]
Hello!

Sorry for the lack of answer, PPC hasn't been at the top of our list due to the lack of an actual machine making testing pretty difficult.
We are currently targeting releasing new package during the month of February.

**Comment by qhaas** [comment]
> Sorry for the lack of answer, PPC hasn't been at the top of our list due to the lack of an actual machine making testing pretty difficult.
> We are currently targeting releasing new package during the month of February.

Thanks for the update.  Those of us that saw promise in the [IBM / nVidia partnership](https://blogs.nvidia.com/blog/2016/09/08/ibm-servers-nvlink/) appreciate the software support.

ppc64le EL7 is still on 1.0.0, hopefully you guys get access to ppc64le hardware soon, interactive development is much easier with podman/docker than Singularity, which we are currently using for GPU acceleration on EL7 ppc64le with our Tesla V100.  Updated libnvidia-container for ppc64le would allow us to switch to podman for GPU acceleration needs.
```
yum info libnvidia-container1 | grep Version
Version     : 1.0.0
```

**Comment by klueska** [comment]
This is the support matrix for the recently released 1.1.0 packages:
```
+----------------------+-------------+----------------+---------+-----------------+
|  OS Name / Version   |  Identifier | amd64 / x86_64 | ppc64le | arm64 / aarch64 |
+======================+=============+================+=========+=================+
| Amazon Linux 1       | amzn1       |       X        |         |                 |
| Amazon Linux 2       | amzn2       |       X        |         |                 |
| Amazon Linux 2017.09 | amzn2017.09 |       X        |         |                 |
| Amazon Linux 2018.03 | amzn2018.03 |       X        |         |                 |
| Open Suse Leap 15.0  | sles15.0    |       X        |         |                 |
| Open Suse Leap 15.1  | sles15.1    |       X        |         |                 |
| Debian Linux 9       | debian9     |       X        |         |                 |
| Debian Linux 10      | debian10    |       X        |         |                 |
| Centos 7             | centos7     |       X        |    X    |                 |
| Centos 8             | centos8     |       X        |    X    |        X        |
| RHEL 7.4             | rhel7.4     |       X        |    X    |                 |
| RHEL 7.5             | rhel7.5     |       X        |    X    |                 |
| RHEL 7.6             | rhel7.6     |       X        |    X    |                 |
| RHEL 7.7             | rhel7.7     |       X        |    X    |                 |
| RHEL 8.0             | rhel8.0     |       X        |    X    |        X        |
| RHEL 8.1             | rhel8.1     |       X        |    X    |        X        |
| RHEL 8.2             | rhel8.2     |       X        |    X    |        X        |
| Ubuntu 16.04         | ubuntu16.04 |       X        |    X    |                 |
| Ubuntu 18.04         | ubuntu18.04 |       X        |    X    |        X        |
| Ubuntu 19.04         | ubuntu19.04 |       X        |    X    |        X        |
| Ubuntu 19.10         | ubuntu19.10 |       X        |    X    |        X        |
| Ubuntu 20.04         | ubuntu20.04 |       X        |    X    |        X        |
+----------------------+-------------+----------------+---------+-----------------+
```

Please let us know if this resolves your issue.

**Comment by qhaas** [comment]
> Please let us know if this resolves your issue.
Thanks!  This will help us with the CUDA / vulkan dependency issues.
```
# yum info libnvidia-container1 | grep Version
Version     : 1.1.1
# cat /etc/redhat-release 
Red Hat Enterprise Linux Server release 7.6 (Maipo)
# lscpu 
Architecture:          ppc64le
Byte Order:            Little Endian
```
